db_stress_tool/db_stress_test_base.cc

```diff
@@ -3635,6 +3635,7 @@ void StressTest::Open(SharedState* shared, bool reopen) {
                 fault_fs_guard->DropRandomUnsyncedFileData(&rand);
               }
             }
+            fault_fs_guard->ResetState();
```

Comment:
This would reset the state of other files created during DB open, including the data written to WAL that are not synced. This may not work when reopen > 0, see the TODO above.

Reply:
This will be an interesting discussion and I'm happy to discuss more:

1. I think the TODO can be removed now since we sync WAL before reopen when FLAGS_reopen > 0 https://github.com/facebook/rocksdb/blob/1f589a3f730c3013b6b80373471b0cd2ae8fde1a/db_stress_tool/db_stress_test_base.cc#L3771-L3790. So if it's agreed, I can remove it and do some rehearsal testing.

2. If we look at ResetState(), it includes `dir_to_new_files_since_last_sync_` and `db_file_state_` (on a side note, I also think it should include `open_managed_files_` so if agreed, I can add that). 

`db_file_state_` is used for (a) reading un-synced data since the recent https://github.com/facebook/rocksdb/commit/9f4c597d839ed9fa9d1a5ed7ea393a150fa9adf8 and (b) knowing what unsync data to be manually dropped with functions DropXXX()

`dir_to_new_files_since_last_sync_` is only used for knowing what files to be manually deleted `DeleteFilesCreatedAfterLastDirSync()`

They are internal states in our testing infra.

3. Because we shared one FS object across different DB opens, these two internal states from last Open can affect the next Open. A problematic way is described in the PR.

I propose to make these internal states stay within each Open so we don't have such affect. Also, I wonder if we should reset the FS object every time we open.... though it's a bit of bigger change so will keep it as follow-up later.

In terms of side effect of such ResetState(), since we manually drop data before we reset the status, the manual deletion purpose of `dir_to_new_files_since_last_sync_` and `db_file_state_` can still be served 

For reading un-synced data, do we expect to read unsync data from last Open in the current Open?? It looks like we does now by coincident but I think it's not expected cc @pdillinger to confirm. If it's not expected, then ResetState() across Open will not defeat serving the purpose of reading un-synced data.

Reply:
1) Yeah, the TODO can be removed (it's actually replaced by your TODO). 

3)  > do we expect to read unsync data from last Open in the current Open

I think this is expected when unsynced data is not dropped.
If we do ResetState() after each DB close, then we are essentially dropping unsynced data after each DB close (it probably does not matter much here since you pointed out that we do WAL sync before doing DB reopen).

Reply:
> (on a side note, I also think it should include open_managed_files_ so if agreed, I can add that).
> the TODO and the parameter reopen can be removed (it's actually replaced by your TODO).

Fixed